pkg/agent/run.go

```diff
@@ -170,7 +175,18 @@ func gatherAndOutputData(ctx context.Context, config Config, preflightClient *cl
 		}
 		log.Println("Data saved locally to", OutputPath)
 	} else {
-		postData(config, preflightClient, readings)
+		//TODO Change backoff parameters to those desired
+		backOff := backoff.NewExponentialBackOff()
+		backOff.MaxElapsedTime = time.Duration(BackoffMaxTime) * time.Second
+		post := func() error {
+			err := postData(config, preflightClient, readings)
+			return err
```

Comment:
This could work as just

```suggestion
			return postData(config, preflightClient, readings)
```

but it doesn't really make a difference.

Reply:
That does look nicer